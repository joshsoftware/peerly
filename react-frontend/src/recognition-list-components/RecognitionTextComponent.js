import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { Form } from "core-components/form/FormComponent";
import ImageComponent from "core-components/image/ImageComponent";

const Border = styled.fieldset`
  border: 1px solid var(--grey);
  border-radius: 11px;
  opacity: 1;
  margin-left: 5%;
  margin-right: 5%;
`;

const Legend = styled.legend`
  width: auto;
  margin-left: 10%;
`;

const ProfileImage = styled(ImageComponent)`
  width: 7vh;
  margin-right: 5%;
`;

const RecognitionTextComponent = ({ text, givenByName, givenByImage }) => (
  <div>
    <Border>
      <Legend>â€œ</Legend>
      <Form.Label className="px-2 pb-2 pt-0">{text}</Form.Label>
    </Border>
    <div className="d-flex justify-content-end w-100">
      <Form.Label className="font-weight-bold text-dark">
        {givenByName}
      </Form.Label>
      <ProfileImage src={givenByImage} roundedCircle={true} alt="Profile" />
    </div>
  </div>
);

RecognitionTextComponent.propTypes = {
  text: PropTypes.string.isRequired,
  givenByName: PropTypes.string.isRequired,
  givenByImage: PropTypes.string,
};

RecognitionTextComponent.defaultProps = {
  givenByImage: "https://i.picsum.photos/id/2/200/200.jpg",
};

export default React.memo(RecognitionTextComponent);
