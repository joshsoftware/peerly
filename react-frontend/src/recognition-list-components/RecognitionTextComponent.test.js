import React from "react";
import { render } from "@testing-library/react";

import RecognitionTextComponent from "recognition-list-components/RecognitionTextComponent";

describe("Recognition text Component test", () => {
  test("should equal snapshot", () => {
    const { asFragment } = render(
      <RecognitionTextComponent givenByName="Roy" text="give high five" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
