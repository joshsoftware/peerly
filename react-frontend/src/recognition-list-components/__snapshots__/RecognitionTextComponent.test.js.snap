// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recognition text Component test should equal snapshot 1`] = `
<DocumentFragment>
  .c0 {
  border: 1px solid var(--grey);
  border-radius: 11px;
  opacity: 1;
  margin-left: 5%;
  margin-right: 5%;
}

.c1 {
  width: auto;
  margin-left: 10%;
}

.c2 {
  width: 7vh;
  margin-right: 5%;
}

<div>
    <fieldset
      class="c0"
    >
      <legend
        class="c1"
      >
        â€œ
      </legend>
      <label
        class="px-2 pb-2 pt-0 form-label"
      >
        give high five
      </label>
    </fieldset>
    <div
      class="d-flex justify-content-end w-100"
    >
      <label
        class="font-weight-bold text-dark form-label"
      >
        Roy
      </label>
      <img
        alt="Profile"
        class="c2 img-fluid rounded-circle"
        src="https://i.picsum.photos/id/2/200/200.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
